packages/core/src/driver.ts

```diff
@@ -405,6 +406,14 @@ class QueryConfig<T = EagerResult> {
      * @type {BookmarkManager|null}
      */
     this.bookmarkManager = undefined
+
+    /**
+     * Configuration for all transactions started to execute the query.
+     *
+     * @type {TransactionConfig|undefined}
+     *
+     */
+    this.transactionConfig = undefined
```

Comment:
TS noob question incoming: are `XYZ|null` and `XYZ|undefined` interchangeable union types? It seems to be mixed here.

Reply:
It can be null actually, but I'm setting undefined since it means the configuration is not set. 

In the case of the `bookmarkManager`, configuring it with `null` means disabling, which is different of configuring it with `undefined` or omitting the configuration.

Reply:
So if it makes a difference, should `bookmarkManager` actually be typed `{BookmarkManager|null|undefined}`?

Reply:
`?`enables the omit the configuration or configure with undefined. In the of `bookmarkManager`, we have defined as `?`and `BookmarkManager | null`. 

The null is not be accepted in typescript for `transactionConfig`, but in javascript, it will get the same behaviour as `undefined` since it's kind of expected.

Reply:
I now understand how the function/constructor types are correct. But what about the comments? `@type {BookmarkManager|null}` vs `@type {TransactionConfig|undefined}`

Reply:
yes, we should type it as `BookmarkManager|null|undefined`.

Reply:
I will fix it.